openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -2166,6 +2189,95 @@ The following is a non-normative example of a Credential Response containing a C
+
+A Wallet MAY provide key attestations to inform the Credential Issuer about the properties of the provided cryptographic public keys, e.g. for proof types sent in the Credential Request. Credential Issuers may want to evaluate these key attestations to determine whether the keys meet their own security requirements, based on the trust framework in use, regulatory requirements, laws, or internal design decisions. A Credential Issuer SHOULD communicate this requirement to evaluate key attestations through its metadata or using some sort of out-of-band mechanism.
+
+Since the key attestations may have large audience as many Credential Issuers that not necessarily uses the same trust framework or internal design decisions, it is required to use a common approach to facilitate interoperability. Therefore, key attestations SHOULD use a common format, allowing Credential Issuers to develop consistent evaluation processes, reducing complexity and potential errors. Common formats makes easy for Credential Issuers to demonstrate compliance with regulatory requirements across different jurisdictions, they also facilitate the development of shared best practices and security benchmarks.
+
+There are two flows how key attestations can be used within Credential issuance:
+- The Wallet uses the `jwt` proof type in the Credential Request to create a proof of possession of the key and adds the key attestation in the JOSE header as specified in (#jwt-proof-type).
+- The Wallet uses the `attestation` proof type in the Credential Request with the key attestation without a proof of possession of the key itself as specified in (#attestation-proof-type).
+
+The latter may avoid unnecessary user interaction during the Credential issuance, as the key itself is not performing a signature operation.
+## Key Attestation in JWT format {#keyattestation-jwt}
+
+The JWT is signed by the Wallet Provider or the Wallet's key storage component itself and contains the following elements:
+
+* in the JOSE header,
+  * `alg`: REQUIRED. A digital signature algorithm identifier such as per IANA "JSON Web Signature and Encryption Algorithms" registry [@IANA.JOSE.ALGS]. It MUST NOT be `none` or an identifier for a symmetric algorithm (MAC).
+  * `typ`: REQUIRED. MUST be `keyattestation+jwt`, which explicitly types the key proof JWT as recommended in Section 3.11 of [@!RFC8725].
+
+The key attestation may use `x5c`, `kid` or `trust_chain` (as defined in (#jwt-proof-type) ) to convey the public key and the associated trust mechanism to sign the key attestation.
+
+* in the JWT body,
+  * `iat`: REQUIRED (number). Integer for the time at which the key attestation was issued using the syntax defined in [@!RFC7519].
+  * `exp`: REQUIRED (number). Integer for the time at which the key attestation and the key(s) it is attesting expire, using the syntax defined in [@!RFC7519].
```

Comment:
in JWT proof type, we only defined iat, and did not define exp, because we said it is up to the issuer's policy to decide how long to accept the proofs. why do we need exp here?

Reply:
the assurances about a key storage component and its keys usually have an expiration, see recommendations from BSI and NIST

Reply:
Yeah I would agree this is a bit different: The key attestation might not come from the wallet directly, but a third party (the wallet provider). The wallet provider should be able to signal (via expiration) how long such an expiration should be deemed valid..
I am wondering if there are there cases where we think it should be optional instead of required?

Reply:
ok to resolve?

Reply:
i think it makes sense for pre-generated key attestations in jwt proof type - it can also be a signal when a wallet needs to ask wallet backend to re-generate the key attestation. but i think it makes less sense in key attestations in attestation proof type, that are generated fresh, with issuer-provided nonce.

Reply:
Thinking with security-by-design I would like to let everything expire

Reply:
Maybe make it OPTIONAL?

There is a `status` claim defined below so I guess the attestation generated by the Wallet Provider could be revocable if it is meant to be long lived or it could use the `exp` otherwise to limit the lifetime of the attestation.

Reply:
yes!

Reply:
I am not opposing but just for my curiosity why does it have to be a MUST for the attestation proof type, is it considered more sensitive?